### ISSUE TITLE
A confusion when reading your paper

### ISSUE THREAD

**Issue by zhiqiangdon** [idiap/importance-sampling#14]
Hi,

Thanks for your awesome work!

For Figure 3 in your paper, the test error curve of 'uniform'  is below that of Loshchilov & Hutter (2015). However, in your analysis in Section 4.2, you said 'The results are depicted in figure 3. We observe that in
the relatively easy CIFAR10 dataset,  all methods can provide some speedup over uniform sampling'. There is a conflict between Figure 3 and this analysis.

For Figure 6, the curve of 'uniform' test error curve almost matches your 'upper-bound' one and it beats other compared methods. 

I am a little confused. The 'uniform' refers to the basic routine of training neural networks. (uniformly sampling data into mini-batch). Why does it beat other compared methods? I think other methods are also designed to improve this 'uniform' baseline. But the results show the contrary. Could you please give some explanations?

**Comment by angeloskath** [comment]
Hi,

Thanks for taking the time and for your good words! I will address your questions one by one:

1. Regarding Figure 3 you are correct that the comment should be _almost_ all methods provide some speedup.
2. In Figure 6 for reasons of clarity (8 lines in a graph are hard to read) we only compare with SVRG type methods. Although they are designed to improve upon the uniform baseline, a nice take-away message from the paper is that when it comes to highly non-convex large model training they actually perform worse than plain SGD with momentum.

Regarding further explanations with respect to the results in Figure 6, the SVRG based methods require two gradient evaluations per parameter update and a certain number of gradient evaluations every `m` iterations. The benefits are reaped mostly at the very final stages of training where almost zero variance is needed. But in the case of large models such as the WRN-28-2 that is used in the paper the overhead is way too much and the constant factors overwhelm the asymptotic improvement. In literature there is barely any comparison of SVRG methods with momentum SGD precisely because momentum SGD sets a very competitive baseline.

I hope I covered your concerns. Feel free to ask more either here or by email.

Cheers,
Angelos

**Comment by zhiqiangdon** [comment]
Thanks for your patient explanation! @angeloskath

**Comment by angeloskath** [comment]
You 're welcome. I am closing the issue, feel free to reopen it if you have further questions or shoot me an email.

**Comment by borre47** [comment]
Hi, what is your email, please?
